receiver/jaegerreceiver/factory.go

```diff
@@ -59,33 +63,48 @@ func (f *Factory) Type() string {
 	return typeStr
 }
 
-// CustomUnmarshaler returns nil because we don't need custom unmarshaling for this config.
+// CustomUnmarshaler is used to add defaults for named but empty protocols
 func (f *Factory) CustomUnmarshaler() receiver.CustomUnmarshaler {
-	return nil
+	return func(v *viper.Viper, viperKey string, intoCfg interface{}, sourceViperSection *viper.Viper) error {
+		// first load the config normally
+		err := sourceViperSection.UnmarshalExact(intoCfg)
+		if err != nil {
+			return err
+		}
+
+		receiverCfg, ok := intoCfg.(*Config)
+		if !ok {
+			return fmt.Errorf("config type not *jaegerreceiver.Config")
+		}
+
+		// next manually search for protocols in viper that do not appear in the normally loaded config
+		// these protocols were excluded during normal loading and we need to add defaults for them
+		vSub := v.Sub(viperKey)
+		if vSub == nil {
+			return fmt.Errorf("Jaeger receiver config is empty")
+		}
+		protocols := vSub.GetStringMap("protocols")
```

Comment:
I think it will be nice to define "protocols" as a const somewhere in `config.go` near `Config struct` to make it clear that this is the key of `Protocols` field and corresponds to the mapstructure tag.